editor/src/components/canvas/canvas-strategies/escape-hatch-strategy.spec.browser2.tsx

```diff
@@ -339,6 +339,247 @@ describe('Convert to Absolute/runEscapeHatch action', () => {
       `),
     )
   })
-  /* eslint-disable @typescript-eslint/no-empty-function */
-  xit('Converts element with absolute descendants', async () => {})
+  it('Converts element with absolute descendants, the absolute descedants are moved to keep their visual position to the new containing block', async () => {
+    const renderResult = await renderTestEditorWithCode(
+      makeTestProjectCodeWithSnippet(`
+      <div
+        style={{
+          position: 'absolute',
+          width: 100,
+          height: 200,
+          left: 10,
+          top: 20,
+        }}
+        data-uid='aaa'
+      >
+        <div
+          style={{
+            backgroundColor: '#FF00BFAB',
+            width: 80,
+            height: 80,
+          }}
+          data-uid='bbb'
+        />
+        <div
+          style={{
+            display: 'flex',
+            padding: 8,
+            flexDirection: 'column',
+          }}
+          data-uid='ccc'
+        >
+          <div style={{ display: 'flex' }} data-uid='ddd'>
+            <div
+              data-uid='eee'
+              style={{
+                position: 'absolute',
+                width: 100,
+                top: 218,
+                height: 20,
+                left: 42,
+              }}
+            >
+              Hello Text
+            </div>
+            <div
+              style={{
+                flexBasis: 100,
+                height: 150,
+                backgroundColor: '#6CF8C0',
+              }}
+              data-uid='fff'
+            />
+          </div>
+        </div>
+      </div>`),
+      'await-first-dom-report',
+    )
+
+    const targets = [EP.appendNewElementPath(TestScenePath, ['aaa', 'ccc'])]
+    await renderResult.dispatch([runEscapeHatch(targets)], true)
+
+    expect(getPrintedUiJsCode(renderResult.getEditorState())).toEqual(
+      makeTestProjectCodeWithSnippet(`
+        <div
+          style={{
+            position: 'absolute',
+            width: 100,
+            height: 200,
+            left: 10,
+            top: 20,
+          }}
+          data-uid='aaa'
+        >
+          <div
+            style={{
+              backgroundColor: '#FF00BFAB',
+              width: 80,
+              height: 80,
+            }}
+            data-uid='bbb'
+          />
+          <div
+            style={{
+              display: 'flex',
+              padding: 8,
+              flexDirection: 'column',
+              position: 'absolute',
+              left: 0,
+              width: 100,
+              top: 80,
+              height: 166,
+            }}
+            data-uid='ccc'
+          >
+            <div style={{ display: 'flex' }} data-uid='ddd'>
+              <div
+                data-uid='eee'
+                style={{
+                  position: 'absolute',
+                  width: 100,
+                  top: 138,
```

Comment:
it's very very hard to see that the actual expected change is this one line!
I wonder how we could improve these tests. 
maybe the test should independently assert the props.style.top of aaa/ccc/ddd/eee is 218 at the beginning and becomes 138 after the dispatch